Automattic/jetpack
# Add pre-publish panel to promote Yoast

## Proposed changes:

* Add pre-publish panel to yoast-promo package
* Initialize and activate yoast-promo package in jetpack plugin
* Address feedback from https://github.com/Automattic/jetpack/pull/29603

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Checkout this branch `add/yoast-promo-prepublish-panel`
* Run your docker env
* Go to "Posts > Add new"
* Click on "Publish" and you should see the below ad (focused on Yoast free version) in the pre-publish panel:
<img width="289" alt="CleanShot 2023-03-22 at 16 56 56@2x" src="https://user-images.githubusercontent.com/8419292/227057043-00bdaa4e-e28f-4e94-aaf2-0a922a191f81.png">

* Ensure CTA link is `https://yoa.st/yoast-jetpack-boost?domain=<your_site>`

* Go to "/wp-admin/plugins.php" and install (and activate) Yoast plugin
* Go to "Posts > Add new"
* Click on "Publish" and you should see the below ad (focused on Yoast premium version) in the pre-publish panel:
<img width="287" alt="CleanShot 2023-03-22 at 17 00 59@2x" src="https://user-images.githubusercontent.com/8419292/227057540-40410a5b-cdfd-4f64-96ac-fcf2144de0b9.png">

* Ensure CTA link is `https://yoa.st/yoast-upgrade-jetpack-boost?domain=<your_site>`
* Click on "x" button - the ad should be dismissed and should not appear again after page reload (the setting is stored in local storage)
* Remove `jetpack-yoast-promo-dismissed` local storage item, the ad should appear again
* A12s - reach out to me to install Yoast Premium plugin
* Go to "Posts > Add new"
* The ad should not be displayed anymore
